src/client/modules/content/widgets/data_table/ProjectCollaboratorTable.jsx

```diff
@@ -0,0 +1,200 @@
+                    organizationCollaborators.push({
+                        name: organization,
+                        read: organizationsWithAccess[organization].read,
+                        write: organizationsWithAccess[organization].write,
+                        delete: organizationsWithAccess[organization].delete,
+                        inOrg: organizationsWithAccess[organization].inOrg
+                    });
+                });
+
+                this.setState({
+                    userCollaborators: userCollaborators.sort(sortObjectArrayByField('name')),
+                    organizationCollaborators: organizationCollaborators.sort(sortObjectArrayByField('name'))
+                });
+            }
+
+        });
+    }
+
+    handleOrderEntries() {
+        if (this.state.display === 1) {
+            this.setState({
+                userCollaborators: this.state.numTimesClicked % 2 === 0 ?
+                    this.state.userCollaborators.sort(sortObjectArrayByField('name')).reverse() :
+                    this.state.userCollaborators.sort(sortObjectArrayByField('name')),
+                numTimesClicked: this.state.numTimesClicked + 1
+            });
+        } else {
+            this.setState({
+                organizationCollaborators: this.state.numTimesClicked % 2 === 0 ?
+                    this.state.organizationCollaborators.sort(sortObjectArrayByField('name')).reverse() :
+                    this.state.organizationCollaborators.sort(sortObjectArrayByField('name')),
+                numTimesClicked: this.state.numTimesClicked + 1
+            });
+        }
+    }
+
+    handleRevoke(event) {
+        this.props.restClient.projects.removeRightsToProject(this.props.ownerId,
+            this.props.projectName,
+            event.target.id)
+            .then(() => {
+                this.retrieveCollaborators(); // Re-render after revoking rights
+            });
+    }
+
+    handleTableSwitch(event) {
+        let newDisplayNum = event.target.innerHTML === 'Users' ? 1 : 2;
+
+        if (newDisplayNum !== this.state.display) {
+            this.setState({
+                display: newDisplayNum
+            });
+        }
+    }
+
+    render() {
+
+        let dataTableData = {
+            categories: {
+                users: [
+                    {id: 1, name: 'UserID:'},
+                    {id: 2, name: 'Rights (RWD)'}
+                ],
+                organizations: [
+                    {id: 1, name: 'OrganizationID:'},
+                    {id: 2, name: 'Rights(RWD)'}
+                ]
+            },
+            dualTable: {
+                show: true,
+                options: ['Users', 'Organizations']
+            }
+        };
+
+        return (
+            <div>
+
+                <DataTable categories={dataTableData.categories.users}
+                           display={this.state.display}
+                           dualTable={dataTableData.dualTable}
```

Comment:
I think either way works - letting the table widgets define box gives more flexibility but less code reuse (not by much) but building it into the table would have less flexibility and more abstraction

But in terms of reusability I guess flexibility is really important so the widgets defining box is probably better for the long term